compiler/lib/src/main/scala/codegen/CppWriter/ComputeCppFiles.scala

```diff
@@ -47,7 +50,8 @@ object ComputeCppFiles extends AstStateVisitor {
     val name = s.getName(Symbol.Component(aNode))
     val loc = Locations.get(node.id)
     for {
-      s <- addMappings(s, FileNames.getComponent(name), Some(loc))
+      s <- if s.template then addMappings(s, FileNames.getComponentImpl(name), Some(loc))
```

Comment:
I think we should have two different visitors here, one for each mode (autocode generation, template generation). Also I think CppWriter should have two different visitors. In autocode gen mode, we should have the same behavior as before. In template gen mode, we should generate only component implementation templates and nothing else (e.g. no types, ports, etc.). We should add tests to the `fpp-filenames` tool to make sure that ComputeCppFiles is working as intended.